Ra 11 burger recipes screen

DenisGubernatorov opened the pull request:
RA-11-Burger-Recipes-Screen 
1) Подготовка данных
2) RecipesListFragment изменен способ обращение к arguments
3) Верстка шапки RecipesListFragment
4) Реализован RecyclerView для списка рецептов
5) Добавлен Fragment для рецепта. Пока пустой
6) Реализован переход на фрагмент рецепта

DenisGubernatorov committed:
RA-11-Burger-Recipes-Screen

1) Подготовка данных
2) RecipesListFragment изменен способ обращение к arguments

DenisGubernatorov committed:
RA-11-Burger-Recipes-Screen

1) Верстка шапки RecipesListFragment
2) Реализован RecyclerView для списка рецептов
3) Добавлен Fragment для рецепта. Пока пустой
4) Реализован переход на фрагмент рецепта

aGroupoid commented on app/src/main/java/com/example/recipesapp/CategoriesListFragment.kt:
```diff
@@ -62,6 +62,7 @@ class CategoriesListFragment : Fragment() {
         }
 
         parentFragmentManager.commit {
+            hide(this@CategoriesListFragment)
```
В целом, этот параметр лишний. Если тебе нужно скрыть его View, то лучше просто использовать replace вместо add

DenisGubernatorov replied:
Что-то я "потерялся" :)) Здесь изначально был replace. Но в https://github.com/DenisGubernatorov/RecipesApp/pull/7#discussion_r1867961651 попросил поменять на add. А если здесь использовать add, то было видно предыдущий фрагмент под новым. Поэтому и добавил скрытие.

aGroupoid commented on app/src/main/java/com/example/recipesapp/CategoriesListFragment.kt:
```diff
@@ -62,6 +62,7 @@ class CategoriesListFragment : Fragment() {
 
         parentFragmentManager.commit {
+            hide(this@CategoriesListFragment)
             add(R.id.mainContainer, recipesListFragment)
```
В прошлых заданиях мы уже использовали более новый котлиновский метод через дженерик (`add<MyFragment>`), давай и дальше продолжать использовать его

aGroupoid commented on app/src/main/java/com/example/recipesapp/RecipesListFragment.kt:
```diff
@@ -23,21 +27,65 @@ class RecipesListFragment : Fragment() {
+            null
+        }
+        binding.recipesListHeaderImg.setImageDrawable(drawableImage)
+        binding.recipesListHeaderImg.contentDescription = "Изображение для категории $categoryName"
```
Хардкодить текстовые параметры - плохая практика. При смене языка, эта строчка не поменяется. Нужно вынести в строковой ресурсы с параметром.
Здесь описано про это
https://developer.android.com/guide/topics/resources/string-resource?hl=ru#formatting-strings

aGroupoid commented on app/src/main/java/com/example/recipesapp/RecipeListAdapter.kt:
```diff
@@ -0,0 +1,60 @@
+                null
+            }
+            binding.recipeImageView.setImageDrawable(drawableImage)
+            binding.recipeImageView.contentDescription = "Изображение для категории ${recipe.title}"
```
Та же проблема с хард код строкой

aGroupoid commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,7 +1,10 @@
     <string name="text_favorites">Избранное</string>
     <string name="text_main_fragment">Основной фрагмент</string>
     <string name="text_recipes_list_title">Список рецептов в категории</string>
+    <string name="recipes_list_img_description">Категория еще в разработке</string>
```
Тестовые поля, как раз лучше не выносить в ресурсы и просто хард-кодить

DenisGubernatorov replied:
Неправильный текст. Это для обязательного атрибута layout/recipes_list_fragment.xml:18

DenisGubernatorov committed:
RA-11-Burger-Recipes-Screen

1) Перевод commit на более современный подход с дженериками
2) Вынос текста для формирования contentDescription для разных фрагментов в ресурсы

aGroupoid reviewed (approved):
По заданию верно, молодец! По поводу наложения, лучше использовать просто add, как я писал раньше. Ты используешь add, но параметров hide, всё откатывается обратно к replace реализации, тогда уж лучше напрямую писать replace.
Скинь в следующем ПР, как это наложение выглядит. Возможно рутовым вьюхам фрагмента достаточно прописать backgroundColor
